library/datasort.c

```diff
@@ -1157,16 +1160,16 @@ static int datasort_swap_memory(struct datasort_cfg *dcfg)
 	 * over it and it can still be used anywhere in code.
 	 */
 	list_replace(&unsorted_bctl->base_entry, &sorted_bctl->base_entry);
-	for (n = 1; n < dcfg->bctl_cnt; ++n)
-		__list_del(dcfg->bctl[n]->base_entry.prev, dcfg->bctl[n]->base_entry.next);
+	for (n = 1; n < dctl->cfg->bctl_cnt; ++n)
```

Comment:
Too much copy-paste of `dctl->cfg` in whole the diff. We should create alias at each function beginning: `struct datasort_cfg *dcfg = dctl->cfg;` and use it. If you do in the this way, your diff will be less.

Reply:
Ok, I'll add it.

Reply:
3b3c9d8ca14403f9c58c48931e98079027ddd8b4